src/jit/lower.cpp

```diff
@@ -491,7 +491,63 @@ void Lowering::DecomposeNode(GenTreePtr* pTree, Compiler::fgWalkData* data)
         }
         break;
     case GT_NEG:
-        NYI("GT_NEG of TYP_LONG");
+        {
+            GenTree* op1 = tree->gtGetOp1();
+            noway_assert(op1->OperGet() == GT_LONG);
+
+            // TODO-Cleanup: Identical code exists in DecomposeStoreInd,
+            // it should be moved to a function similar to CreateTemporary
+            GenTreeStmt* newStmt = comp->fgInsertEmbeddedFormTemp(&op1->gtOp.gtOp1);
+            if (newStmt->gtStmtIsTopLevel())
+            {
+                for (GenTreeStmt* nextEmbeddedStmt = newStmt->gtStmtNextIfEmbedded();
+                     nextEmbeddedStmt != nullptr;
+                     nextEmbeddedStmt = nextEmbeddedStmt->gtStmt.gtStmtNextIfEmbedded())
+                {
+                    comp->compCurStmt = nextEmbeddedStmt;
+                    comp->fgWalkTreePost(&nextEmbeddedStmt->gtStmt.gtStmtExpr, Lowering::DecompNodeHelper, this, true);
+                }
+
+                comp->compCurStmt = curStmt;
+            }
+
+            newStmt = comp->fgInsertEmbeddedFormTemp(&op1->gtOp.gtOp2);
+            assert(!newStmt->gtStmtIsTopLevel());  
```

Comment:
That logic is only needed if the newly added statement is a top level statement instead of an embedded statement. This can only happen if the tree that's stored in a temporary starts right at the beginning of the current statement. So, once a temporary is created there's no way that subsequent temporary creation could produce a new top level statement, the `GT_LCL_VAR` node created for the first temporary is in the way.

There's an assert there to make sure my understanding of embedded statements is correct, I've been bitten to many times by embedded statement issues.

You can see similar code in other lowering code, `LowerArrElem` for example. Once we create a function the same logic will probably always be used, event though it's unnecessary in some cases.